zap/src/main/java/org/zaproxy/zap/spider/parser/SpiderHtmlParser.java

```diff
@@ -47,6 +47,37 @@ public class SpiderHtmlParser extends SpiderParser {
             Pattern.compile(
                     "(?:http(?:s?):)?//[^\\x00-\\x1f\"'\\s<>#()\\[\\]{}]+",
                     Pattern.CASE_INSENSITIVE);
+    /**
+     * Functional interface that allows custom handling of URLs retrieved in attributes, to be
+     * manipulated before the url is sent to a {@link SpiderParser:processURL} call.
+     */
+    @FunctionalInterface
+    private interface CustomUrlProcessor {
+        void process(HttpMessage message, int depth, String localURL, String baseURL);
+    }
+
+    /**
+     * Implements the CustomUrlProcessor to handle the special format of a srcset attribute element,
+     * see https://developer.mozilla.org/en-US/docs/Web/HTML/Element/img#attr-srcset the srcset
+     * attribute contains one or more strings separated by commas, indicating possible image sources
+     * for the user agent to use.
+     */
+    private class SrcSetProcessor implements CustomUrlProcessor {
+
+        @Override
+        public void process(HttpMessage message, int depth, String localURL, String baseURL) {
+            // Regexp to split entries for srcsets: [^"\'=\s,]+\.([^\s,]+)
+            // 1) Matches a single character not present in [^"\'=\s,] multiple times
+            // 2) Matches a .
+            // 3) Matches a single character not present in [^\s,] multiple times
+            Matcher results = Pattern.compile("[^\"'=\\s,]+\\.([^\\s,]+)").matcher(localURL);
+            while (results.find()) {
+                if (!results.group().isEmpty()) {
+                    processURL(message, depth, results.group(), baseURL);
```

Comment:
results.group is the entire match right? So the group inside the regex can be marked as non-capturing? (I believe that has a slight performance benefit.)

Reply:
Yes, I had it as a capturing group because I wanted initially to check also for a set of possible filetypes, but I deemed it not necessary. Fixing it accordingly

Reply:
👍